src/simulation/elements/PSTN.cpp

```diff
@@ -325,25 +336,28 @@ static int MoveStack(Simulation * sim, int stackX, int stackY, int directionX, i
 	} else {
 		StackData stackData = CanMoveStack(sim, stackX, stackY, directionX, directionY, maxSize, amount, retract, block);
 		int currentPos = stackData.pushed + stackData.spaces;
-		if(currentPos){
-			//Move particles
-			int possibleMovement = 0;
-			for(int j = currentPos-1; j >= 0; j--) {
-				int jP = tempParts[j];
-				if(jP < 0) {
-					possibleMovement++;
-					continue;
+		if (stackData.spaces>=minextend)
```

Comment:
Watch the spacing here and on the next line. Always put a space between operators and after if statements. You should probably combine these, too.
```
if (currPos && stackData.spaces >= minextend)
{
```